lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -53,7 +53,11 @@ def add_key
   end
 
   def add_key_from_key_server
-    command = "gpg --keyserver #{resource[:key_server]} --recv-keys #{resource[:key_id]}"
+    if resource[:proxy].empty?
```

Comment:
In my environment, I had to set this to .nil?  With .empty? I received the following:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Error: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp
Error: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp
Wrapped exception:
undefined method `empty?' for nil:NilClass
Error: /Stage[main]/Rvm::Gnupg_key/Gnupg_key[rvm_D39DC0E3]/ensure: change from absent to present failed: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp

After setting to "if resource[:proxy].nil?" receive the below:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Debug: Executing 'gpg --keyserver hkp://keys.gnupg.net --keyserver-options http-proxy=http://myproxy.company.com:80 --recv-keys D39DC0E3'

ruby 2.0.0p598 (2014-11-13) [x86_64-linux]
puppet 3.8.7